sesman/scp_v1.c

```diff
@@ -156,66 +151,126 @@ scp_v1_process(struct SCP_CONNECTION *c, struct SCP_SESSION *s)
-                sitem = session_get_bypid(sid);
-
-                if (0 == sitem)
-                {
-                    e = scp_v1s_connection_error(c, "Internal error");
-                    LOG(LOG_LEVEL_INFO, "Cannot find session item on the chain");
-                }
-                else
-                {
-                    display = sitem->display;
-                    /*e=scp_v1s_reconnect_session(c, sitem, display);*/
-                    e = scp_v1s_reconnect_session(c, display);
-
-                    if (0 != s->client_ip)
-                    {
-                        LOG(LOG_LEVEL_INFO, "++ reconnected session: username %s, display :%d.0, session_pid %d, ip %s", s->username, display, sitem->pid, s->client_ip);
-                    }
-                    else
-                    {
-                        LOG(LOG_LEVEL_INFO, "++ reconnected session: username %s, display :%d.0, session_pid %d", s->username, display, sitem->pid);
-                    }
+    /* cleanup */
+    if (do_auth_end)
+    {
+        auth_end(data);
+    }
+    g_free(slist);
+    return SCP_SERVER_STATE_OK;
+}
 
-                    g_free(sitem);
-                }
+/******************************************************************************/
+static enum SCP_SERVER_STATES_E
+scp_v1_process4(struct trans *t, struct SCP_SESSION *s)
+{
+    return SCP_SERVER_STATE_OK;
+}
 
-                break;
-            default:
-                /* we check the other errors */
-                parseCommonStates(e, "scp_v1s_list_sessions()");
-                break;
-        }
+/******************************************************************************/
+static enum SCP_SERVER_STATES_E
+scp_v1_process41(struct trans *t, struct SCP_SESSION *s)
+{
+    int scount;
+    enum SCP_SERVER_STATES_E e;
+    struct SCP_DISCONNECTED_SESSION *slist;
+
+    /* list disconnected sessions */
+    slist = session_get_byuser(s->username, &scount,
+                               SESMAN_SESSION_STATUS_DISCONNECTED);
+
+    if (scount == 0)
+    {
+        /* */
+        return SCP_SERVER_STATE_END;
     }
 
-    /* resource management */
-    if ((e == SCP_SERVER_STATE_OK) && (s->rsr))
+    e = scp_v1s_list_sessions42(t, scount, slist);
+    if (SCP_SERVER_STATE_OK != e)
     {
-        /* here goes scp resource sharing code */
+        LOG(LOG_LEVEL_ERROR, "scp_v1s_list_sessions42 failed");
     }
 
-    /* cleanup */
-    if (do_auth_end)
+    return SCP_SERVER_STATE_OK;
+}
+
+/******************************************************************************/
+static enum SCP_SERVER_STATES_E
+scp_v1_process43(struct trans *t, struct SCP_SESSION *s)
+{
+    struct session_item *sitem;
+    enum SCP_SERVER_STATES_E e;
+    int display;
+
+    sitem = session_get_bypid(s->return_sid);
+    if (0 == sitem)
     {
-        auth_end(data);
+        e = scp_v1s_connection_error(t, "Internal error");
+        LOG(LOG_LEVEL_INFO, "Cannot find session item on the chain");
```

Comment:
What's the "chain"? can you please make this log message more informative, like "The server does not have a session with id %s"?

Reply:
Agreed.